lib/contracts/accountParser.js

```diff
@@ -0,0 +1,65 @@
+const bip39 = require("bip39");
+const hdkey = require('ethereumjs-wallet/hdkey');
+const fs = require('../core/fs');
+
+class AccountParser {
+  static parseAccountsConfig(accountsConfig, web3, logger) {
+    let accounts = [];
+    if (accountsConfig && accountsConfig.length) {
+      accountsConfig.forEach(accountConfig => {
+        const account = AccountParser.getAccount(accountConfig, web3, logger);
+        if (!account) {
+          return;
+        }
+        if (Array.isArray(account)) {
+          accounts = accounts.concat(account);
+          return;
+        }
+        accounts.push(account);
+      });
+    }
+    return accounts;
+  }
+
+  static getAccount(accountConfig, web3, logger) {
+    if (!logger) {
+      logger = console;
+    }
+    if (accountConfig.privateKey) {
+      if (!accountConfig.privateKey.startsWith('0x')) {
+        accountConfig.privateKey = '0x' + accountConfig.privateKey;
+      }
+      return web3.eth.accounts.privateKeyToAccount(accountConfig.privateKey);
+    }
+    if (accountConfig.privateKeyFile) {
+      let fileContent = fs.readFileSync(fs.dappPath(accountConfig.privateKeyFile)).toString();
```

Comment:
Consider using this instead. https://web3js.readthedocs.io/en/1.0/web3-eth-accounts.html#decrypt